apps/dev/src/commands/decoder/generate-decoder/generate-decoder.ts

```diff
@@ -35,28 +34,53 @@ export const generateDecoder = Command.make(
       Options.withDefault('generated-decoders'),
     ),
   },
-  ({ decoderType, evmVersion, outputDir, witFunction, witPath, witWorld }) =>
+  ({
+    decoderType,
+    evmVersion,
+    outputDir,
+    stride,
+    witFunction,
+    witPath,
+    witWorld,
+  }) =>
     Effect.gen(function* () {
       const witFilePath = path.join(rootDir, witPath);
       if (!(yield* Effect.tryPromise(() => fs.stat(witFilePath)))) {
         return yield* Effect.fail(`WIT file not found at path: ${witFilePath}`);
       }
 
+      // Generate temporary JSON representation from WIT file
+      const tempOutputPath = path.join(rootDir, 'tmp/wit-output.json');
+      yield* Effect.tryPromise(() =>
+        fs.mkdir(path.dirname(tempOutputPath), { recursive: true }),
+      );
+
       const args = [
         '--input',
         path.join(rootDir, witPath),
+        '--output',
+        tempOutputPath,
         '--world',
         witWorld,
         '--function',
         witFunction,
       ];
-      const res = yield* Effect.tryPromise(() =>
-        execPromise(`wit-converter ${args.join(' ')}`, {
-          cwd: path.join(rootDir, 'apps/wit-converter'),
-        }),
+
+      yield* Exec.make('wit-converter', ...args).pipe(
+        Exec.workingDirectory(path.join(rootDir, 'apps/wit-converter')),
+        Exec.string,
+        Effect.provide(NodeContext.layer),
```

Comment:
The NodeContext is provided in `apps/dev/src/bin.ts` so you don't need to provide it here too:
```suggestion
```